src/install/install.zig

```diff
@@ -5086,85 +5086,77 @@ pub const PackageManager = struct {
 
-                if (strings.hasPrefix("http://", request.name) or
-                    strings.hasPrefix("https://", request.name))
-                {
-                    if (Output.isEmojiEnabled()) {
-                        Output.prettyErrorln("<r>😢 <red>error<r><d>:<r> bun {s} http://url is not implemented yet.", .{
-                            @tagName(op),
-                        });
-                    } else {
-                        Output.prettyErrorln("<r><red>error<r><d>:<r> bun {s} http://url is not implemented yet.", .{
-                            @tagName(op),
-                        });
-                    }
-
+                const placeholder = String.from("@@@");
+                var version = Dependency.parseWithOptionalTag(
+                    allocator,
+                    placeholder,
+                    value,
+                    null,
+                    &SlicedString.init(value, value),
+                    log,
+                ) orelse switch (op) {
+                    .link, .unlink => null,
+                    else => brk: {
+                        value = std.fmt.allocPrint(allocator, "npm:{s}", .{value}) catch unreachable;
+                        break :brk Dependency.parseWithOptionalTag(
+                            allocator,
+                            placeholder,
+                            value,
+                            null,
+                            &SlicedString.init(value, value),
+                            log,
+                        );
+                    },
+                } orelse {
+                    Output.prettyErrorln("<r><red>error<r><d>:<r> unrecognised dependency format: {s}", .{
+                        positional,
+                    });
                     Global.exit(1);
+                };
+                switch (version.tag) {
+                    .dist_tag => if (version.value.dist_tag.name.eql(placeholder, value, value)) {
+                        value = std.fmt.allocPrint(allocator, "npm:{s}", .{value}) catch unreachable;
+                        version = Dependency.parseWithOptionalTag(
+                            allocator,
+                            placeholder,
+                            value,
+                            null,
+                            &SlicedString.init(value, value),
+                            log,
+                        ) orelse {
+                            Output.prettyErrorln("<r><red>error<r><d>:<r> unrecognised dependency format: {s}", .{
+                                positional,
+                            });
+                            Global.exit(1);
+                        };
+                    },
+                    else => {},
                 }
 
-                request.name = std.mem.trim(u8, request.name, "\n\r\t");
-                if (request.name.len == 0) continue;
-
+                var request = UpdateRequest{
+                    .name = switch (version.tag) {
```

Comment:
this is a pointer to a temporary which becomes invalid after this stack frame (scope)

Reply:
`String` is temporary, yes, but it is immediately `.slice(value)` into `[] u8`, and `value` is allocated on the heap AFAICT...? :thinking:

Reply:
Done in https://github.com/oven-sh/bun/pull/1895/commits/cd3aa41a203b0876ee9ab9423856d8833af18140